packages/vscode-extension/lib/wrapper.js

```diff
@@ -190,53 +221,47 @@ export function AppWrapper({ children, initialProps, ..._rest }) {
             x: frame.left / width,
             y: frame.top / height,
             width: frame.width / width,
-            height: frame.height / height,
+            height: frame.height / height
           };
-          let stackPromise = Promise.resolve(undefined);
-          if (payload.requestStack) {
-            stackPromise = Promise.all(
-              viewData.hierarchy.reverse().map((item) => {
-                const inspectorData = item.getInspectorData((arg) => findNodeHandle(arg));
-                const framePromise = new Promise((resolve, reject) => {
-                  try {
-                    inspectorData.measure((_x, _y, viewWidth, viewHeight, pageX, pageY) => {
-                      resolve({
-                        x: pageX / width,
-                        y: pageY / height,
-                        width: viewWidth / width,
-                        height: viewHeight / height,
-                      });
-                    });
-                  } catch (e) {
-                    reject(e);
-                  }
-                });
-
-                return framePromise
-                  .catch(() => undefined)
-                  .then((frame) => {
-                    return inspectorData.source
-                      ? {
-                          componentName: item.name,
-                          source: {
-                            fileName: inspectorData.source.fileName,
-                            line0Based: inspectorData.source.lineNumber - 1,
-                            column0Based: inspectorData.source.columnNumber - 1,
-                          },
-                          frame,
-                        }
-                      : undefined;
-                  });
-              })
-            ).then((stack) => stack?.filter(Boolean));
-          }
-          stackPromise.then((stack) => {
+          
+          if (!requestStack) {
             devtoolsAgent._bridge.send("RNIDE_inspectData", {
               id: payload.id,
               frame: scaledFrame,
-              stack: stack,
             });
-          });
+            return;
+          }
+
+          const inspectNode = (node, stack) => {
+            // Optimization: we break after reaching fiber node corresponding to OffscreenComponent (with tag 22).
+            if (!node || node.tag === 22) {
```

Comment:
why is it 22? Is it documented somewhere or at least there's some code that sets it we can link to?

I recall that there's some numbering convention for root view instances, but from what I remember it changes in between JS reloads (it gets incremented) and also used to be different between Android and iOS. Can you check if this works well after JS reload too?

Reply:
From what I see, the tag does not change between reloads. What changes is the unique node identifier, in some places it's called `_nativeTag`, in other `handle`.

These constants are defined here: https://github.com/facebook/react/blob/c3570b158d087eb4e3ee5748c4bd9360045c8a26/packages/react-reconciler/src/ReactWorkTags.js#L62

I will add this link to the comment